taichi/backends/dx/dx_device.cpp

```diff
@@ -0,0 +1,464 @@
+#include <unordered_map>
+
+#include "taichi/backends/dx/dx_device.h"
+#include "taichi/backends/dx/dx_api.h"
+#include "spirv_hlsl.hpp"
+
+namespace taichi {
+namespace lang {
+namespace directx11 {
+
+extern IDXGISwapChain *g_swapchain;
+
+static uint32_t g_alloc_id = 0;
+
+// Mapping from binding ID to Buffer
+static std::unordered_map<uint32_t, ID3D11Buffer *> g_binding2buf;
+
+// Mapping from binding ID to CPU-accessible copy of the resource
+static std::unordered_map<uint32_t, ID3D11Buffer *> g_binding2cpucopy;
+
+// Mapping from binding ID to UAV
+static std::unordered_map<uint32_t, ID3D11UnorderedAccessView *> g_binding2uav;
+
+DxResourceBinder::~DxResourceBinder() {
+}
+
+std::unique_ptr<ResourceBinder::Bindings> DxResourceBinder::materialize() {
+  TI_NOT_IMPLEMENTED;
+  return nullptr;
+}
+
+void DxResourceBinder::rw_buffer(uint32_t set,
+                                 uint32_t binding,
+                                 DevicePtr ptr,
+                                 size_t size) {
+  TI_NOT_IMPLEMENTED;
+}
+
+void DxResourceBinder::rw_buffer(uint32_t set,
+                                 uint32_t binding,
+                                 DeviceAllocation alloc) {
+  TI_ASSERT_INFO(set == 0, "DX only supports set = 0, requested set = {}", set);
+  ID3D11UnorderedAccessView *uav = g_binding2uav[alloc.alloc_id];
+  binding_map_[binding] = uav;
+}
+
+void DxResourceBinder::buffer(uint32_t set,
+                              uint32_t binding,
+                              DevicePtr ptr,
+                              size_t size) {
+  TI_NOT_IMPLEMENTED;
+}
+
+void DxResourceBinder::buffer(uint32_t set,
+                              uint32_t binding,
+                              DeviceAllocation alloc) {
+  rw_buffer(set, binding, alloc);
```

Comment:
`Buffer` means constant buffer in DX / uniform buffer in Vulkan. Buffer should not be bind as an RWByteAccessBuffer

Reply:
It looks this function has never been called through the test cases I had run, so I'm setting this to "not implemented".